gateway/transport/extensions/runbookhook.go

```diff
@@ -93,40 +96,138 @@ func getRunbookHookFiles(ctx Context) (*runbookWrapperFiles, error) {
+	var hooks []*runbookWrapperFiles
+	runbooksConfig, err := models.GetRunbookConfigurationByOrgID(models.DB, ctx.OrgID)
+	if err != nil {
+		if err == models.ErrNotFound {
+			return hooks, nil
+		}
+		return nil, err
+	}
+
+	if cached, ok := hookStoreV2.Load(ctx.OrgID); ok {
+		cachedHooks := cached.([]*runbookWrapperFiles)
+		now := time.Now().UTC()
+
+		hasExpired := slices.ContainsFunc(cachedHooks, func(hook *runbookWrapperFiles) bool {
+			return now.After(hook.expireAt)
+		})
+		if !hasExpired {
+			log.With("sid", ctx.SID).Infof("loading runbook hook V2 from cache")
+			return cachedHooks, nil
+		}
+
+		hookStoreV2.Delete(ctx.OrgID)
+	}
+
+	// Fetch hooks from repositories
+	for _, repositoryConfig := range runbooksConfig.RepositoryConfigs {
+		config, err := runbooks.NewConfig(repositoryConfig)
+		if err != nil {
+			return nil, err
+		}
+
+		if config.HookCacheTTL == nil {
+			return hooks, nil
+		}
+
+		repository, err := runbooks.FetchRepository(config)
+		if err != nil {
+			return nil, err
+		}
+
+		sessionOpenF, err := repository.ReadFile(sessionOpenHookFileName, map[string]string{})
+		if err != nil {
+			return nil, err
+		}
+		sessionCloseF, err := repository.ReadFile(sessionCloseHookFileName, map[string]string{})
+		if err != nil {
+			return nil, err
+		}
+		f := &runbookWrapperFiles{
+			sessionOpen:   sessionOpenF,
+			sessionClose:  sessionCloseF,
+			expireAt:      time.Now().UTC().Add(*config.HookCacheTTL),
+			cacheDuration: *config.HookCacheTTL,
+		}
+
+		hooks = append(hooks, f)
+	}
+	hookStoreV2.Store(ctx.OrgID, hooks)
+
+	return hooks, nil
+}
+
+func executeOpenSessionRunbookHook(ctx Context, pkt *proto.Packet, hook *runbookWrapperFiles) {
+	requestID := uuid.NewSHA1(uuid.NameSpaceURL, fmt.Appendf(nil, "session-open/%s", ctx.SID)).String()
+	resp := transportsystem.RunRunbookHook(ctx.AgentID, &pbsystem.RunbookHookRequest{
+		ID:        requestID,
+		SID:       ctx.SID,
+		Command:   []string{hookDefaultCommand},
+		InputFile: string(hook.sessionOpen.InputFile),
+		EventSessionOpen: &pbsystem.EventSessionOpen{
+			Verb:                ctx.Verb,
+			ConnectionName:      ctx.ConnectionName,
+			ConnectionType:      ctx.ConnectionType,
+			ConnectionSubType:   ctx.ConnectionSubType,
+			ConnectionEnvs:      ctx.ConnectionEnvs,
+			ConnectionReviewers: ctx.ConnectionReviewers,
+			Input:               string(pkt.Payload),
+			UserEmail:           ctx.UserEmail,
+		},
+	})
+	log.With("sid", ctx.SID).Infof("session-open: runbook hook finished, duration=%v, exit-code=%v, output=%v",
+		resp.ExecutionTimeSec, resp.ExitCode, resp.Output)
+}
+
 func processEventOpenSessionHook(ctx Context, pkt *proto.Packet) {
 	hook, err := getRunbookHookFiles(ctx)
 	if err != nil {
 		log.With("sid", ctx.SID).Warnf("session-open: failed fetching runbook hook, reason=%v", err)
 		return
 	}
-	if hook == nil || hook.sessionOpen == nil {
+	hooksV2, err := getRunbookHookFilesV2(ctx)
```

Comment:
If the user has an old configuration, how this would be handled? It's a breaking change?